blueprints/app/files/.eslintrc.js

```diff
@@ -57,10 +57,32 @@ module.exports = {
         node: true,
       },
       extends: ['plugin:n/recommended'],
-    },
+    },<% if (templateTag) { %>
+    // Template tag files
+    {
+      files: ['**/*.gjs'],
+      parser: 'ember-eslint-parser',
+      plugins: ['ember'],
+      extends: [
+        'eslint:recommended',
+        'plugin:ember/recommended',
+        'plugin:ember/recommended-gjs',
+      ],
+    },<% if (typescript) { %>
+    {
+      files: ['**/*.gts'],
+      parser: 'ember-eslint-parser',
+      plugins: ['ember'],
+      extends: [
+        'eslint:recommended',
+        'plugin:@typescript-eslint/recommended',
+        'plugin:ember/recommended',
+        'plugin:ember/recommended-gts',
+      ],
+    },<% } %><% } %>
     {
       // test files
-      files: ['tests/**/*-test.{js,ts}'],
+      files: ['tests/**/*-test.{js,ts<% if (templateTag) { %>,gjs,gts<% } %>}'],
```

Comment:
no need for if statement (imo, obvs, but strongly), this should just be in the default.